src/twisted/web/newsfragments/9547.bugfix

```diff
@@ -0,0 +1 @@
+FileBodyProducer will now stop producing when cancelled.  This behavior is relied on by twisted.web._newclient.Request.
```

Comment:
- In news files, we should always qualify python identifiers that have changed - so `FileBodyProducer` should be `twisted.web.client.FileBodyProducer`
- The qualifications should always be by their *public* alias; so `twisted.web._newclient.Request` should be `twisted.web.client.Request`.  (I realize that it's annoying to figure out what the "public alias" of something defined in a private implementation module is, and that's why I wrote https://pypi.org/project/publication/ and I hope we can adopt it in Twisted one day :))
- `FileBodyProducer` doesn't itself have a `cancel` method, so it might be confusing to folks to say "when cancelled" here; "the `Deferred` returned by `FileBodyProducer.startProducing`" might be clearer.
- What does it mean that this is "relied upon" by `...client.Request`?

Reply:
Thanks!

> What does it mean that this is "relied upon" by ...client.Request?
I suppose it means that `...client.Request` has a method that seems to expect that calling `cancel()` on the `Deferred` which was returned from `FileBodyProducer.startProducing()` will actually do something.  Perhaps this isn't relevant here, so I just removed the sentence.